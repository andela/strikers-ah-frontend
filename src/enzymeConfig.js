import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-localstorage-mock';

export default Enzyme.configure({
  adapter: new Adapter(),
});
