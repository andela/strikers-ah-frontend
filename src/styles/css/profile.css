@import url('https://fonts.googleapis.com/css?family=Open+Sans&display=swap');
*,
*::before,
*::after {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Roboto Condensed', sans-serif;
}

body {
  background: #f0f0f0;
  font-family: 'Open Sans', sans-serif;
}

@keyframes showAddImage {
  from {
    background: transparent;
    opacity: 0;
  }
  to {
    background: #0788f1;
    opacity: 0.7;
  }
}

@keyframes modal {
  from {
    margin-top: 0;
    border-radius: 0;
    box-shadow: none;
  }
  to {
    margin-top: 3%;
    border-radius: 10px;
    box-shadow: 0px 0px 14px #333;
  }
}

.web-heading {
  color: #079bfd;
}

.profile-grid-off,
.profile-grid {
  padding: 2%;
  position: relative;
}
.profile-grid-off .user-account-info,
.profile-grid .user-account-info {
  padding: 1.5%;
  box-shadow: 0px 0px 14px rgba(191, 191, 191, 0.24);
  border-radius: 10px;
  background: white;
}
.profile-grid-off .user-account-info .statistics,
.profile-grid .user-account-info .statistics {
  width: 100%;
  border-top: 2px solid #ddd;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  text-align: center;
  margin-top: 1%;
  padding-top: 1%;
}
.profile-grid-off .user-account-info .statistics .stat-label,
.profile-grid .user-account-info .statistics .stat-label {
  text-decoration: none;
  color: #949292;
  display: inline-block;
}
.profile-grid-off .user-account-info .statistics .stat-label .value,
.profile-grid .user-account-info .statistics .stat-label .value {
  color: #333;
  font-weight: bold;
  font-size: 1.2em;
}
.profile-grid-off .user-account-info .user-account-flex,
.profile-grid .user-account-info .user-account-flex {
  position: relative;
  display: grid;
  grid-template-columns: 10em 1fr;
  grid-column-gap: 10px;
}
.profile-grid-off .user-account-info .user-account-flex .img-container,
.profile-grid .user-account-info .user-account-flex .img-container {
  position: relative;
  height: 10em;
  width: 10em;
  margin-right: 2em;
  text-align: center;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .img-container
  .add-img-toggler,
.profile-grid
  .user-account-info
  .user-account-flex
  .img-container
  .add-img-toggler {
  position: absolute;
  background: #ddd;
  bottom: 0;
  border: none;
  outline: none;
  left: 0;
  width: 100%;
  background: transparent;
  font-weight: bolder;
  color: transparent;
  font-size: 2em;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .img-container
  .add-img-toggler:hover,
.profile-grid
  .user-account-info
  .user-account-flex
  .img-container
  .add-img-toggler:hover {
  animation-name: showAddImage;
  animation-duration: 0.5s;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  color: white;
}
.profile-grid-off .user-account-info .user-account-flex .img-container img,
.profile-grid .user-account-info .user-account-flex .img-container img {
  width: 100%;
  height: 100%;
  cursor: pointer;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .img-container
  img:hover
  ~ .add-img-toggler,
.profile-grid
  .user-account-info
  .user-account-flex
  .img-container
  img:hover
  ~ .add-img-toggler {
  z-index: 1;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .img-container:hover
  .profile-image-label,
.profile-grid
  .user-account-info
  .user-account-flex
  .img-container:hover
  .profile-image-label {
  display: inline-block;
}
.profile-grid-off .user-account-info .user-account-flex .info-container,
.profile-grid .user-account-info .user-account-flex .info-container {
  padding: 1em 0;
  grid-area: 'info-container';
}
.profile-grid-off .user-account-info .user-account-flex .info-container .name,
.profile-grid .user-account-info .user-account-flex .info-container .name {
  font-weight: 500;
  font-size: 2em;
  text-transform: capitalize;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .info-container
  .username,
.profile-grid .user-account-info .user-account-flex .info-container .username {
  line-height: 40px;
  color: #949292;
}
.profile-grid-off .user-account-info .user-account-flex .info-container .bio,
.profile-grid .user-account-info .user-account-flex .info-container .bio {
  padding: 1% 10px 1% 0px;
  line-height: 30px;
  color: #535353;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .info-container
  .btn-follow,
.profile-grid
  .user-account-info
  .user-account-flex
  .info-container
  .btn-follow {
  float: right;
  margin-right: 10px;
  border: 1px solid #079bfd;
  width: 100px;
  padding: 5px;
  border-radius: 5px;
  color: #079bfd;
  margin-top: 10px;
  outline: none;
  font-size: 0.9em;
  transition: 0.2s;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .info-container
  .btn-follow:hover,
.profile-grid
  .user-account-info
  .user-account-flex
  .info-container
  .btn-follow:hover {
  box-shadow: 0px 0px 14px rgba(191, 191, 191, 0.24);
  background: #079bfd;
  color: white;
  transform: translateY(-2px);
}
.profile-grid-off .upload-image-modal,
.profile-grid .upload-image-modal {
  position: absolute;
  width: 100%;
  height: 70vh;
  top: 0;
  left: 50%;
  background: rgba(86, 86, 87, 0.7);
  transform: translateX(-50%);
  z-index: 100;
}
.profile-grid-off .upload-image-modal .upload-page,
.profile-grid .upload-image-modal .upload-page {
  position: relative;
  width: 90%;
  margin-top: 3%;
  margin-left: auto;
  margin-right: auto;
  background: #eff2f5;
  min-height: 90%;
  box-shadow: 0px 0px 14px #333;
  border-radius: 10px;
  animation-name: modal;
  animation-duration: 0.3s;
}
.profile-grid-off .upload-image-modal .btn,
.profile-grid .upload-image-modal .btn {
  font-size: 18px;
  line-height: 18px;
  height: 30px;
  padding: 0 10px;
  border-radius: 1px;
  margin: 1%;
}
.profile-grid-off .upload-image-modal .btn-cancel,
.profile-grid .upload-image-modal .btn-cancel {
  margin-left: 10%;
  border: 1px solid brown;
  color: #f38b8b;
}
.profile-grid-off .upload-image-modal .btn-cancel:hover,
.profile-grid .upload-image-modal .btn-cancel:hover {
  background: brown;
  color: white;
}
.profile-grid-off .upload-image-modal .btn-upload,
.profile-grid .upload-image-modal .btn-upload {
  background: white;
  border: 1px solid #079bfd;
  color: #079bfd;
}
.profile-grid-off .upload-image-modal .btn-upload:hover,
.profile-grid .upload-image-modal .btn-upload:hover {
  background: #079bfd;
  color: white;
}
.profile-grid-off .upload-image-modal .dropzone,
.profile-grid .upload-image-modal .dropzone {
  width: 100%;
  background: #e1f0ff;
  height: 100px;
  margin-bottom: 1%;
  position: relative;
}
.profile-grid-off .upload-image-modal .dropzone p,
.profile-grid .upload-image-modal .dropzone p {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}
.profile-grid-off .upload-image-modal .crop .crop-section,
.profile-grid .upload-image-modal .crop .crop-section {
  width: 100%;
  display: grid;
  grid-template-columns: 60% 1% 30%;
  grid-column-gap: 1em;
}
.profile-grid-off .upload-image-modal .crop .cropper,
.profile-grid .upload-image-modal .crop .cropper {
  width: 100%;
  overflow: auto;
  height: 400px;
  text-align: center;
  border-radius: 10px;
}
.profile-grid-off .upload-image-modal .crop .separator,
.profile-grid .upload-image-modal .crop .separator {
  background: #dde4e9;
}
.profile-grid-off .upload-image-modal .crop .crop-preview,
.profile-grid .upload-image-modal .crop .crop-preview {
  width: 100%;
  overflow: hidden;
  text-align: center;
}
.profile-grid-off .upload-image-modal .crop .crop-preview canvas,
.profile-grid .upload-image-modal .crop .crop-preview canvas {
  border-radius: 10px;
  box-shadow: 0px 0px 14px #949292;
  border: 5px solid #ddd;
}
.profile-grid-off .left-profile,
.profile-grid .left-profile {
  background: white;
  box-shadow: 0px 0px 14px rgba(191, 191, 191, 0.24);
  border-radius: 10px;
}
.profile-grid-off .left-profile li,
.profile-grid .left-profile li {
  list-style-type: none;
  border-top: 1px solid #f0f0f0;
  line-height: 35px;
  padding-left: 10px;
}
.profile-grid-off .left-profile li .link,
.profile-grid .left-profile li .link {
  text-decoration: none;
  color: #949292;
}
.profile-grid-off .left-profile li .active,
.profile-grid .left-profile li .active {
  color: #079bfd;
}
.profile-grid-off .left-profile h3,
.profile-grid .left-profile h3 {
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
  line-height: 40px;
  padding-left: 10px;
  background: rgba(9, 129, 241, 0.89);
  color: white;
}

.profile-grid {
  display: grid;
  grid-template-columns: 1fr 3fr;
  grid-column-gap: 2em;
  grid-template-areas: 'left-profile user-account-flex';
}

@media screen and (max-width: 1500px) {
  .profile-grid {
    width: 80%;
    margin: 2em auto;
    display: grid;
    grid-template-columns: 1fr 3fr;
    grid-column-gap: 10px;
    margin: 2em auto;
  }
}

@media screen and (max-width: 1000px) {
  .profile-grid {
    display: grid;
    grid-template-columns: 1fr 3fr;
    grid-column-gap: 4px;
    width: 98%;
    margin: 2em auto;
  }
}

@media screen and (max-width: 815px) {
  .profile-grid {
    width: 80%;
    display: grid;
    grid-template-columns: 1fr;
    grid-column-gap: 4px;
    width: 98%;
    margin: 2em auto;
  }
  .profile-grid .left-profile {
    display: none;
  }
}

@media screen and (max-width: 500px) {
  .profile-grid-off {
    display: grid;
    grid-template-columns: 1fr;
    width: 98%;
    margin: 2em auto;
  }
  .profile-grid-off .user-account-flex {
    display: grid;
    grid-template-columns: 1fr;
    grid-column-gap: 0px;
    grid-template-areas: 'image-container' 'info-container';
  }
  .profile-grid-off .user-account-flex .info-container {
    display: block;
    width: 100%;
  }
  .profile-grid-off .user-account-flex .info-container .name {
    font-weight: 100;
    font-size: 1em;
  }
  .profile-grid {
    display: block;
  }
}

.text-center {
  text-align: center;
}
