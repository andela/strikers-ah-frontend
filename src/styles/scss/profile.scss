@import 'common';
@import 'animation';
$color-blue: rgb(7, 155, 253);
$color-brown: brown;
.web-heading {
  color: $color-blue;
}
.profile-grid-off,
.profile-grid {
  position: relative;
  .user-account-info {
    padding: 1.5%;
    box-shadow: 0px 0px 14px $box-shadow-color;
    border-radius: 10px;
    background: $color-white;
    .statistics {
      width: 100%;
      border-top: 2px solid $color-right-gray;
      display: grid;
      grid-template-columns: 1fr 1fr 1fr;
      text-align: center;
      margin-top: 1%;
      padding-top: 1%;
      .stat-label {
        text-decoration: none;
        color: $color-text-gray;
        display: inline-block;
        .value {
          color: $color-black;
          font-weight: bold;
          font-size: 1.2em;
        }
      }
    }
    .user-account-flex {
      display: grid;
      grid-template-columns: 10em 1fr;
      grid-column-gap: 10px;
      .img-container {
        position: relative;
        height: 10em;
        width: 10em;
        margin-right: 2em;
        text-align: center;
        .add-img-toggler {
          position: absolute;
          background: $color-right-gray;
          bottom: 0;
          border: none;
          outline: none;
          left: 0;
          width: 100%;
          background: transparent;
          font-weight: bolder;
          color: transparent;
          font-size: 2em;
          &:hover {
            animation-name: showAddImage;
            animation-duration: 0.5s;
            animation-iteration-count: 1;
            animation-fill-mode: forwards;
            color: white;
          }
        }
        img {
          width: 100%;
          height: 100%;
          cursor: pointer;
          &:hover ~ .add-img-toggler {
            z-index: 1;
          }
        }
        &:hover .profile-image-label {
          display: inline-block;
        }
      }
      .info-container {
        padding: 1em 0;
        grid-area: 'info-container';
        text-align: justify;
        .name {
          font-weight: 500;
          font-size: 2em;
          text-transform: capitalize;
        }
        .username {
          line-height: 40px;
          color: $color-text-gray;
        }
        .bio {
          padding: 1% 10px 1% 0px;
          line-height: 30px;
          color: $color-dark-text-gray;
        }
        .btn-follow {
          margin-left: 2em;
          border: 1px solid $color-blue;
          width: 100px;
          padding: 5px;
          border-radius: 5px;
          color: $color-blue;
          margin-top: 10px;
          outline: none;
          font-size: 0.9em;
          transition: 0.2s;
          &:hover {
            box-shadow: 0px 0px 14px $box-shadow-color;
            background: $color-blue;
            color: $color-white;
            transform: translateY(-2px);
          }
        }
      }
    }
  }
  .upload-image-modal {
    position: absolute;
    width: 100%;
    height: 90vh;
    width: 76vw;
    top: -2em;
    left: -4.5em;
    z-index: 100;
    .upload-page {
      width: 90%;
      margin-top: 3%;
      margin-left: auto;
      margin-right: auto;
      background: rgb(239, 242, 245);
      min-height: 90%;
      box-shadow: 0px 0px 14px #333;
      border-radius: 10px;
      animation-name: modal;
      animation-duration: 0.3s;
    }
    .btn {
      font-size: 18px;
      line-height: 18px;
      height: 30px;
      padding: 0 10px;
      border-radius: 1px;
      margin: 1%;
    }
    .btn-cancel {
      margin-left: 10%;
      border: 1px solid $color-brown;
      color: rgb(243, 139, 139);
      &:hover {
        background: $color-brown;
        color: white;
      }
    }
    .btn-upload {
      background: white;
      border: 1px solid $color-blue;
      color: $color-blue;
      &:hover {
        background: $color-blue;
        color: white;
      }
    }
    .dropzone {
      width: 100%;
      background: rgb(225, 240, 255);
      height: 100px;
      margin-bottom: 1%;
      position: relative;

      p {
        position: absolute;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
      }
    }
    .crop {
      .crop-section {
        width: 100%;
        display: grid;
        grid-template-columns: 60% 1% 30%;
        grid-column-gap: 1em;
      }
      .cropper {
        margin-left: 1%;
        width: 100%;
        overflow: auto;
        height: 400px;
        text-align: center;
        border-radius: 10px;
      }
      .separator {
        background: rgb(221, 228, 233);
      }
      .crop-preview {
        width: 100%;
        overflow: hidden;
        text-align: center;
        canvas {
          border-radius: 10px;
          box-shadow: 0px 0px 14px $color-text-gray;
          border: 5px solid $color-right-gray;
        }
      }
    }
  }
  .left-profile {
    background: $color-white;
    box-shadow: 0px 0px 14px $box-shadow-color;
    border-radius: 10px;
    li {
      list-style-type: none;
      border-top: 1px solid $color-gray-tertiary;
      line-height: 35px;
      padding-left: 10px;
      .link {
        text-decoration: none;
        color: $color-text-gray;
      }
      .active {
        color: $color-blue;
      }
    }
    h3 {
      border-top-left-radius: 10px;
      border-top-right-radius: 10px;
      line-height: 40px;
      padding-left: 10px;
      background: rgba(9, 129, 241, 0.89);
      color: white;
    }
  }
}
.profile-grid {
  display: block;
  width: 100%;
  // grid-template-columns: 1fr 3fr;
  // grid-column-gap: 2em;
  // grid-template-areas: 'left-profile user-account-flex';
}
@media screen and (max-width: 1500px) {
  .profile-grid {
    width: 100%;
    margin: 0 auto 2em auto;
    // display: grid;
    // grid-template-columns: 1fr 3fr;
    // grid-column-gap: 10px;
    margin: 2em auto;
  }
}
@media screen and (max-width: 1000px) {
  .profile-grid {
    // display: grid;
    // grid-template-columns: 1fr 3fr;
    // grid-column-gap: 4px;
    width: 98%;
    margin: 2em auto;
  }
}
@media screen and (max-width: 815px) {
  .profile-grid {
    width: 100%;
    // display: grid;
    // grid-template-columns: 1fr;
    // grid-column-gap: 4px;
    margin: 2em auto;
    .left-profile {
      display: none;
    }
  }
}
@media screen and (max-width: 500px) {
  .profile-grid-off {
    display: grid;
    grid-template-columns: 1fr;
    width: 98%;
    margin: 2em auto;
    .user-account-flex {
      display: grid;
      grid-template-columns: 1fr;
      grid-column-gap: 0px;
      grid-template-areas: 'image-container' 'info-container';
      .info-container {
        display: block;
        width: 100%;
        .name {
          font-weight: 100;
          font-size: 1em;
        }
      }
    }
  }
  .profile-grid {
    display: block;
  }
}
.text-center {
  text-align: center;
}
.friend-profile {
  width: 60%;
  margin-left: auto;
  margin-right: auto;
}
.profile-page {
  position: relative;
}
