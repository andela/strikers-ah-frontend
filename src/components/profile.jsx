import React, { Component } from 'react';
import { connect } from 'react-redux';
// import LeftProfile from './common/leftProfile';
import RightProfile from './common/profileRightSide';
import { getUserProfile, getUserArticles } from '../redux/actions/userAction';
import '../styles/css/profile.css';
import { getLoggedInUser } from '../helpers/authentication';

/**
 * @author Mwibutsa FLoribert
 * @returns { * } --
 */
class Profile extends Component {
  /**
   * @author Mwibutsa Floribert
   * @param {*} props
   */
  constructor(props) {
    super(props);
    this.state = {
      editProfile: { showForm: false, showImageForm: false },
      pagination: { pageSize: 5, currentPage: 0 },
    };
    const {
      getUserProfile: getProfile,
      getUserArticles: getArticles,
    } = this.props;
    const { username } = this.props;
    getProfile(username);
    getArticles(username);
  }

  /**
   * @author Mwibutsa Floribert
   * @param { * } nextPage --
   * @param { * } pages --
   * @param { * } prev --
   * @param { * } next --
   * @returns { * } --
   */
  paginate = (nextPage, pages, prev = false, next = false) => {
    const { pagination } = this.state;
    let activePage = pagination.currentPage;
    if (prev) {
      if (pagination.currentPage === 0) activePage = pages - 1;
      else activePage -= 1;
    } else if (next) {
      if (pagination.currentPage === pages - 1) activePage = 0;
      else activePage += 1;
    } else {
      activePage = nextPage;
    }
    this.setState({ pagination: { ...pagination, currentPage: activePage } });
  };

  toggleEditProfile = () => {
    const { editProfile } = this.state;
    this.setState({ editProfile: { showForm: !editProfile.showForm } });
  };

  toggleShowImage = () => {
    const { editProfile } = this.state;
    this.setState({
      editProfile: { showImageForm: !editProfile.showImageForm },
    });
  };
  /**
   * @author Mwibutsa FLoribert
   * @param { Integer } loggedInUserId --
   * @param { Integer } userProfileId --
   * @returns { Boolean } returns the comparison of logged in user id and user profile
   */

  isOwner = (loggedInUserId, userProfileId) => {
    return loggedInUserId === userProfileId;
  };

  /**
   * @author Mwibutsa FLoribert
   * @returns { JSX } --
   */
  render() {
    const { accountInfo: user } = this.props;
    const { articles } = this.props;
    let userArticles = [];
    if (Object.keys(articles).length) {
      userArticles = articles.articles;
    }
    let { profile } = user;
    profile = profile || user;
    const defaultUser = {
      name: 'Not Provided',
      image:
        'https://cdn.pixabay.com/photo/2013/07/13/10/44/man-157699__480.png',
      bio: '',
      followers: [],
      followings: [],
    };
    const isLoggedIn = !!getLoggedInUser();
    const owner = getLoggedInUser().id === profile.id;
    let ownerClass = 'profile-page';
    if (!isLoggedIn || !owner) {
      ownerClass += ' friend-profile';
    }
    const { editProfile, pagination } = this.state;
    return (
      <div className={ownerClass} test-data="profile-page" id="profile-page">
        <div className="profile-grid">
          {profile.email && (
            <RightProfile
              accountInfo={{
                ...defaultUser,
                ...profile,
                owner,
              }}
              editProfile={{
                showForm: editProfile.showForm,
                toggleEditProfile: this.toggleEditProfile,
                toggleShowImage: {
                  visible: editProfile.showImageForm,
                  toggleForm: this.toggleShowImage,
                },
              }}
              userArticles={userArticles}
              pagination={pagination}
              paginate={this.paginate}
            />
          )}
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => {
  return {
    accountInfo: state.userProfile.user,
    articles: state.userArticles,
  };
};
export default connect(
  mapStateToProps,
  { getUserProfile, getUserArticles },
)(Profile);
