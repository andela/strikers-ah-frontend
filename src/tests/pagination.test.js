import React from 'react';
import '../enzymeConfig';
import { getComponent, findByTestAttribute } from './profile/test-helpers';
import Pagination from '../components/common/pagination';

describe('TEST PAGINATION COMPONENT', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = { pages: 2, activePage: 1, changePage: jest.fn() };
    wrapper = getComponent(<Pagination {...props} />);
  });
  test('Should be able to render pagination', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('On page change', () => {
    const nextLink = wrapper.find('.next');
    nextLink.simulate('click', { preventDefault: jest.fn() });
    const prevLink = wrapper.find('.prev');
    prevLink.simulate('click', { preventDefault: jest.fn() });
    const linksInLoop = findByTestAttribute(wrapper, 'linksInLoopComponent').at(
      0,
    );
    linksInLoop.simulate('click', { preventDefault: jest.fn() });
    expect(props.changePage).toHaveBeenCalledTimes(3);
  });
  test('When page size equal to pages', () => {
    props.pages = 1;
    wrapper = getComponent(<Pagination {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
