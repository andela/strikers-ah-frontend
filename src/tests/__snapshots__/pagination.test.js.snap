// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST PAGINATION COMPONENT Should be able to render pagination 1`] = `ShallowWrapper {}`;

exports[`TEST PAGINATION COMPONENT When page size equal to pages 1`] = `ShallowWrapper {}`;
